components/Details.tsx

```diff
@@ -1,27 +1,27 @@
 import React from "react";
 import PropTypes from "prop-types";
 
+interface Summary {
+  id: string;
+  value: string | React.ReactNode;
+}
+
 interface Props {
-  summary:
-    | string
-    | {
-        id: string;
-        value: string;
-      };
+  summary: React.ReactNode | Summary;
   children: React.ReactNode;
 }
 
 export const Details = (props: Props): JSX.Element => {
   const { summary, children } = props;
 
   let summaryId: string | undefined = undefined;
-  let summaryValue: string;
+  let summaryValue: string | React.ReactNode;
 
-  if (typeof summary === "string") {
+  if ((summary as Summary).id && (summary as Summary).value) {
+    summaryId = (summary as Summary).id;
+    summaryValue = (summary as Summary).value;
```

Comment:
You can make this a little simpler by casting and assigning. So `const s = summary as Summary`. Not a requirement, though.